File: benchmark/c/cpw/total_number_of_non_decreasing_numbers_with_n_digits_1/total_number_of_non_decreasing_numbers_with_n_digits_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int total_number_of_non_decreasing_numbers_with_n_digits_1 ( int n ) ;
int total_number_of_non_decreasing_numbers_with_n_digits_1 ( int n ) {
  int N = 10;
  long long count = 1;
  for ( int i = 1;
  i <= n;
  i ++ ) {
    count *= ( N + i - 1 );
    count /= i;
  }
  return count;
}

File: translation_result/else_bms/fail/c/cpw/total_number_of_non_decreasing_numbers_with_n_digits_1/gemini/Hinted/total_number_of_non_decreasing_numbers_with_n_digits_1.rs

fn total_number_of_non_decreasing_numbers_with_n_digits_1(n: i32) -> i32 {
    let n = n as usize;
    let mut count = 1;
    for i in 1..=n {
        count = (count * (10 + i - 1)) / i;
    }
    count as i32
}
